src/contracts/SavingCircles.sol

```diff
@@ -0,0 +1,276 @@
+   * @param _token Address of a token
+   * @return bool Token allowed
+   */
+  function isTokenAllowed(address _token) external view override returns (bool) {
+    return allowedTokens[_token];
+  }
+
+  /**
+   * @notice Return the info of a specified saving circle
+   * @param _id Identifier of the circle
+   * @return _circle Saving circle
+   */
+  function circle(bytes32 _id) external view override returns (Circle memory _circle) {
+    _circle = circles[_id];
+
+    if (_isDecommissioned(_circle)) revert NotCommissioned();
+
+    return _circle;
+  }
+
+  /**
+   * @notice Return the balances of the members of a specified saving circle
+   * @param _id Identifier of the circle
+   * @return _members Members of the specified saving circle
+   * @return _balances Corresponding balances of the members of the circle
+   */
+  function memberBalances(bytes32 _id)
+    external
+    view
+    override
+    returns (address[] memory _members, uint256[] memory _balances)
+  {
+    Circle memory _circle = circles[_id];
+
+    if (_isDecommissioned(_circle)) revert NotCommissioned();
+
+    _balances = new uint256[](_circle.members.length);
+    for (uint256 i = 0; i < _circle.members.length; i++) {
+      _balances[i] = balances[_id][_circle.members[i]];
+    }
+
+    return (_circle.members, _balances);
+  }
+
+  /**
+   * @notice Return the member address which is currently able to withdraw from a specified circle
+   * @param _id Identifier of the circle
+   * @return address Member that is currently able to withdraw from the circle
+   */
+  function withdrawableBy(bytes32 _id) external view override returns (address) {
+    Circle memory _circle = circles[_id];
+
+    if (_isDecommissioned(_circle)) revert NotCommissioned();
+
+    return _circle.members[_circle.currentIndex];
+  }
+
+  /**
+   * @notice Return if a circle can currently be withdrawn from
+   * @param _id Identifier of the circle
+   * @return bool If the circle is able to be withdrawn from
+   */
+  function withdrawable(bytes32 _id) external view override returns (bool) {
+    return _withdrawable(_id);
+  }
+
+  /**
+   * @dev Make a deposit into a specified circle
+   *      A deposit must be made in specific time window and can be made partially so long as the final balance equals
+   *      the specified deposit amount for the circle.
+   */
+  function _deposit(bytes32 _id, uint256 _value, address _member) internal {
+    Circle memory _circle = circles[_id];
+
+    if (_isDecommissioned(_circle)) revert NotCommissioned();
+    if (!isMember[_id][_member]) revert NotMember();
+    if (
+      block.timestamp < circles[_id].circleStart
+        || block.timestamp >= circles[_id].circleStart + (circles[_id].depositInterval * (circles[_id].currentIndex + 1))
+        || block.timestamp >= circles[_id].circleStart + (circles[_id].depositInterval * circles[_id].maxDeposits)
+        || balances[_id][_member] + _value > circles[_id].depositAmount
```

Comment:
I'm wondering if allowing flexibility in the deposit amount instead of only allowing the full amount contributes to ux or makes it more confusing.. thoughts @diterra-code ?